### PR TITLE
#234207 - ProcessManagement step prioritization change

### PR DESCRIPTION
**Pull request by VishnuReghuMDSL**
- Fixed the issue where processing the 1st step of the 1st pipeline - the 2nd step of the 1st pipeline goes to the back of the queue (after the 1st step of the 500th pipeline).
- Added a while loop to go through each step in the pipeline before moving to next pipeline, so all steps in that pipeline will be executed first
- If there was an error while executing the step, the while loop was continously looking for the failed step hence all other workflows were not starting as it was waiting for this never ending while loop
- Hence checking for StartTime is null when taking list of second step

**Describe the change**
A clear and concise description of what the change is. Any PR submitted without a description of the change will not be reviewed.

**Describe your implementation or design**
How did you go about implementing the change?

**Tests**
Did you cover your changes with tests?

**Breaking change**
Do you changes break compatibility with previous versions?

**Additional context**
Any additional information you'd like to provide?

### CODE REVIEW

File: src/WorkflowCore/WorkflowCore.csproj [hunk 1]
```diff
@@ -6,15 +6,22 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <AssemblyName>WorkflowCore</AssemblyName>
     <PackageId>WorkflowCore</PackageId>
+	<GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <PackageTags>workflow;.NET;Core;state machine</PackageTags>
     <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
     <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
     <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
     <Description>Workflow Core is a light weight workflow engine targeting .NET Standard.</Description>
+	<PublishNupkgLibraryCommand>nuget.exe push -Source https://intelligentcomms.pkgs.visualstudio.com/_packaging/calero-mdsl/nuget/v3/index.json -ApiKey az</PublishNupkgLibraryCommand>
```
Comment (DaveFuller-Calero) [inline src/WorkflowCore/WorkflowCore.csproj]: Do we actually need the changes that are in this file?
Our github is currently open, so anyone could view these api keys.
If we need this then perhaps we need to make our github private.

File: src/WorkflowCore/MDSL.GitInfo.props [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿<!--
```
Comment (DaveFuller-Calero) [inline src/WorkflowCore/MDSL.GitInfo.props:1]: Do we actually need this file?

**Review by DaveFuller-Calero — COMMENTED** [review]
The main WorkflowExecutor.cs change is what we need, and looks fine (it looks like a lot changed, but it's mostly whitespace changes).
I'm not sure if we need the changes from the other 2 files. I'd like to keep changes to a minimum if these aren't needed, to help avoid conflicts - but if they're required for us I think they're fine.

File: src/WorkflowCore/WorkflowCore.csproj (hunk 1 above)
Comment (VishnuReghuMDSL) [reply src/WorkflowCore/WorkflowCore.csproj]: Removed unnecessary tag. We can always add in local .csproj before pushing the nuget package.

File: src/WorkflowCore/MDSL.GitInfo.props (hunk 2 above)
Comment (VishnuReghuMDSL) [reply src/WorkflowCore/MDSL.GitInfo.props:1]: We will keep this file for versioning as Hydra.STL.NugetPackaging uses this file to version the nuget package.

**Review by DaveFuller-Calero — APPROVED** [review]

**Pull request merged**